types/index.ts

```diff
@@ -450,282 +537,381 @@ Units examples
-  assert.deepStrictEqual(math.floor(math.fraction(-44,7)), math.fraction(-7))
-  assert.deepStrictEqual(math.floor(math.fraction(44,7), 2), math.fraction(628,100))
-  assert.deepStrictEqual(math.floor(math.fraction(44,7), math.bignumber(2)), math.fraction(628,100))
+  assert.deepStrictEqual(math.floor(math.fraction(44, 7)), math.fraction(6))
+  assert.deepStrictEqual(math.floor(math.fraction(-44, 7)), math.fraction(-7))
+  assert.deepStrictEqual(
+    math.floor(math.fraction(44, 7), 2),
+    math.fraction(628, 100)
+  )
+  assert.deepStrictEqual(
+    math.floor(math.fraction(44, 7), math.bignumber(2)),
+    math.fraction(628, 100)
+  )
 
   // Complex input
-  const c = math.complex(3.24, -2.71);
-  assert.deepStrictEqual(math.floor(c), math.complex(3, -3));
-  assert.deepStrictEqual(math.floor(c, 1), math.complex(3.2, -2.8));
-  assert.deepStrictEqual(math.floor(c, math.bignumber(1)), math.complex(3.2, -2.8));
+  const c = math.complex(3.24, -2.71)
+  assert.deepStrictEqual(math.floor(c), math.complex(3, -3))
+  assert.deepStrictEqual(math.floor(c, 1), math.complex(3.2, -2.8))
+  assert.deepStrictEqual(
+    math.floor(c, math.bignumber(1)),
+    math.complex(3.2, -2.8)
+  )
 
   // array input
-  assert.deepStrictEqual(math.floor([3.2, 3.8, -4.7]), [3, 3, -5]);
-  assert.deepStrictEqual(math.floor([3.21, 3.82, -4.71], 1), [3.2, 3.8, -4.8]);
-  assert.deepStrictEqual(math.floor([3.21, 3.82, -4.71], math.bignumber(1)),
-    math.bignumber([3.2, 3.8, -4.8]));
+  assert.deepStrictEqual(math.floor([3.2, 3.8, -4.7]), [3, 3, -5])
+  assert.deepStrictEqual(math.floor([3.21, 3.82, -4.71], 1), [3.2, 3.8, -4.8])
+  assert.deepStrictEqual(
+    math.floor([3.21, 3.82, -4.71], math.bignumber(1)),
+    math.bignumber([3.2, 3.8, -4.8])
+  )
 
   // numeric input, array or matrix of decimals
-  let numFloored: math.MathArray = math.floor(math.tau, [2, 3])
-  assert.deepStrictEqual(numFloored, [6.28, 6.283]);
-  let bigFloored: math.Matrix = math.floor(math.bignumber(6.28318), math.matrix([2, 3]));
-  assert.deepStrictEqual(bigFloored, math.matrix(math.bignumber([6.28, 6.283])));
-  assert.deepStrictEqual(math.floor(math.fraction(44, 7), [2, 3]),
-    [math.fraction(628, 100), math.fraction(6285, 1000)]);
-
-  // array - array is (currently) undefined
-  //@ts-expect-error
-  assert.throws(() => math.floor([3.21, 3.82], [1, 2]), TypeError);
+  const numFloored: math.MathArray = math.floor(math.tau, [2, 3])
+  assert.deepStrictEqual(numFloored, [6.28, 6.283])
+  const bigFloored: math.Matrix = math.floor(
+    math.bignumber(6.28318),
+    math.matrix([2, 3])
+  )
+  assert.deepStrictEqual(bigFloored, math.matrix(math.bignumber([6.28, 6.283])))
+  assert.deepStrictEqual(math.floor(math.fraction(44, 7), [2, 3]), [
+    math.fraction(628, 100),
+    math.fraction(6285, 1000),
+  ])
+
+  // @ts-expect-error ... Should throw runtime error, array - array is (currently) undefined
+  assert.throws(() => math.floor([3.21, 3.82], [1, 2]), TypeError)
```

Comment:
Did Prettier really reorder comments in this way? I think it's more confusing now. The @ts-expect-error has nothing to do with the runtime error; it tells tsc that the following line is malformed TypeScript, and more importantly for the testing sake of index.ts, it _enforces_ that the following line should be a TypeScript error. If the definitions were changed so that the following line would type properly, then tsc would throw an "unused expect error" error. It's the only way I could find to make certain that I had not made the type declarations too permissive. The fact that it happens then to also be a runtime error is in some sense coincidental, and handled separately, by the `assert.throws()` on the next line. So to me glomming the two comments on to one line obscures the fact that they are about two different things.

Reply:
Prettier did not re-order the comments. Apparently `plugin:@typescript-eslint/recommended` includes a rule [`ban-ts-comment`](https://typescript-eslint.io/rules/ban-ts-comment/) that requires a description for `@ts-expect-error`. The description has to be on the same line as the ts-comment, though, so the original comment
```ts
  // array - array is (currently) undefined
  // @ts-expect-error
```
didn't count.

I'll remove the part about runtime and just revert to what you originally had. What I was trying to express with that was...The assert is asserting a runtime Type Error is thrown, but the error can't be triggered without a @ts-expect-error, since ts will catch the TypeError at compile time.

Reply:
@ChristopherChudzicki I think you still wanted to change this comment again? It still is: 

```js
// @ts-expect-error ... array - array is (currently) undefined
```

right now which indeed looks a bit confusing

Reply:
I agree this comment is confusing. But I'm also confused on what we want here.

The current comment is just the original comment, but moved onto same line as the `ts-expect-error` that it was explaining, which appeases the linter.  I **think** this comment is saying "`math.floor` does not support arrays right now. It's going to throw a compile-time type error and a runtime type error".

Anyway, since I'm not entirely sure of the meaning / desired behavior, keeping the original comment (added in #2539) but appeasing the linter seemed like a good option.

If you or @gwhitney have a suggested wording, I'm happy to update the PR with that.

Reply:
Maybe `@ts-expect-error ... verify the undefined floor(array, array) does not type`

If that's clearer. The point is I added this line because in the first pass of this typing, TypeScript was letting this through and allowing the runtime error to occur, when one of the main points of typescriptis to catch as much as possible at compile time. So I made the typing more restrictive and added this test to make sure I had really caught the situation at compile time.

Or leaving it as it is now is also fine with me.